src/routers/books.js

```diff
@@ -1,4 +1,62 @@
-// Import data here...
+const express = require("express");
+const router = express.Router();
+const { books } = require("../../data");
+const {
+  checkForMissingFields,
+  checkForExistingEntry,
+  findById,
+} = require("../../src/utils/utils");
 
+// GET
 
-// Write routes here...
+router.get("/", (req, res) => {
+  // Gets all books
+  res.json({ books });
+});
+
+router.get("/:id", (req, res) => {
+  res.json({ book: findById(books, Number(req.params.id)) });
+});
+
+// POST
+
+router.post("/", (req, res) => {
+  checkForMissingFields(books, req.body);
+  checkForExistingEntry(books, req.body);
+  const newBook = { id: books.length + 1, ...req.body };
+  books.push(newBook);
+  res.status(201).json({ book: newBook });
+});
+
+// DELETE
+
+router.delete("/:id", (req, res) => {
+  const foundBook = findById(books, Number(req.params.id));
+  const indexOfFoundBook = books.indexOf(foundBook);
+  books.splice(indexOfFoundBook, 1);
+  res.json({ book: foundBook });
+});
+
+// PUT
+
+router.put("/:id", (req, res) => {
+  const foundBook = findById(books, Number(req.params.id));
+  checkForExistingEntry(books, req.body);
+  const indexOfFoundBook = books.indexOf(foundBook);
+  const updatedBook = req.body;
+  books[indexOfFoundBook] = { ...foundBook, ...updatedBook };
```

Comment:
I love the way you're handling these routes but this one likely has a bug. If the client were to ever attach an `id` to the JSON body, that would update the ID of the record as well. I'd add something into this controller that removes an ID field from the body if there is one to be sure